SimulationAgent/DeviceConnection/DeviceConnectionActor.cs

```diff
@@ -312,6 +360,18 @@ public void HandleEvent(ActorEvents e)
                     this.status = ActorStatus.Done;
                     break;
 
+                case ActorEvents.Disconnected:
+                    this.actorLogger.DeviceDisconnected();
+                    // TODO: this works for the time being, but disconnection should not always lead to a deregistration
+                    //       e.g. there are simulation scenarios where the device might just need to disconnect
```

Comment:
I added this TODO for the future - I would have expected the deletion to be more explicit and not depending on a disconnection -- we can fix this later